test/permissions/remote.provider.getDetails.js

```diff
@@ -0,0 +1,214 @@
+      login(users.S2.ReadOnlyUser, () =>
+        remoteProvider.getDetails(
+          'Permission',
+          permissions.S2.ReadOnlyCategory,
+          'Actions',
+          isError(test)
+        )
+      )
+  );
+
+  test.test(
+    'remoteProvider.getDetails: check read permission, no read permission for category in Many decorator',
+    (test, { remoteProvider, login }) =>
+      login(users.S1.ReadOnlyUser, () =>
+        remoteProvider.getDetails(
+          'Permission',
+          permissions.S1.ReadOnlyCategory,
+          'Actions',
+          isError(test)
+        )
+      )
+  );
+
+  test.test(
+    'remoteProvider.getDetails: check read permission for category with subsystem/catalog, sufficient permission',
+    (test, { remoteProvider, login }) =>
+      login(users.S1.AdminUser, () =>
+        remoteProvider.getDetails(
+          'SystemUser',
+          users.S1.AdminUser,
+          'Roles',
+          isSuccess(test)
+        )
+      )
+  );
+
+  test.test(
+    'remoteProvider.getDetails: check read permission for category with subsystem/catalog, no record in category',
+    (test, { remoteProvider, login }) =>
+      login(users.S1.AdminUser, () =>
+        remoteProvider.getDetails(
+          'SystemUser',
+          roles.S1.Admin,
+          'Roles',
+          isError(test)
+        )
+      )
+  );
+
+  test.test(
+    'remoteProvider.getDetails: check read permission, no read permission for category with subsystem/catalog',
+    (test, { remoteProvider, login }) =>
+      login(users.S2.ReadOnlyUser, () =>
+        remoteProvider.getDetails(
+          'SystemUser',
+          users.S2.ReadOnlyUser,
+          'Roles',
+          isError(test)
+        )
+      )
+  );
+
+  test.test(
+    'remoteProvider.getDetails: check read permission, no read permission for subsystem/catalog',
+    (test, { remoteProvider, login }) =>
+      login(users.S1.ReadOnlyUser, () =>
+        remoteProvider.getDetails(
+          'SystemUser',
+          users.S2.ReadOnlyUser,
+          'Roles',
+          isError(test)
+        )
+      )
+  );
+
+  test.test(
+    'remoteProvider.getDetails: check read permission, no read permission for category with subsystem/catalog in Many decorator',
+    (test, { remoteProvider, login }) =>
+      login(users.S1.ReadOnlyUser, () =>
+        remoteProvider.getDetails(
+          'SystemUser',
+          users.S1.ReadOnlyUser,
+          'Roles',
+          (err, res) => {
+            test.error(err);
+            test.assertNot(res[0]);
+            test.end();
+          }
+        )
+      )
+  );
+
+  test.test(
+    'remoteProvider.getDetails: check read permission, no read permission for subsystem/catalog in Many decorator',
+    (test, { remoteProvider, login }) =>
+      login(users.S1.AdminUser, () =>
+        provider.linkDetails(
+          'SystemUser',
+          'Roles',
+          users.S1.GuestUser,
+          roles.S2.Guest,
+          err => {
```

Comment:
Nit: you can use `test.cb` in such cases, it will do the `test.error(err)` for you.

Reply:
Thanks, haven't noticed this